doc/arch/010-dagger-qualifier-migration.md

```diff
@@ -0,0 +1,55 @@
+# ADR 010: Dagger Qualifier Migration
+## Status:
+Accepted on 2020-08-05
+
+## Context:
+We have been using `@Named` annotation to provide dependencies that have same return type.
+```kotlin
+// Providing dependency
+@Provides
+@Named("full_date")
+fun provideDateFormatterForFullDate(locale: Locale): DateTimeFormatter = DateTimeFormatter.ofPattern("d-MMM-yyyy", locale)
+
+@Provides
+@Named("date_for_user_input")
+fun provideDateFormatterForUserInput(locale: Locale): DateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy", locale)
+
+// Injecting dependency
+@Inject
+@Named("full_date")
+private lateinit var dateFormatter: DateTimeFormatter
+```
+
+This is problematic because this is not a type-safe approach and prone to developer error like having a typo in `@Named` value when providing and injecting the dependency, we can find the issue after building the project when Dagger graph is generated.
```

Comment:
This sentence is too long, can we break this up into smaller chunks, please?

Reply:
Done. I have still left the problem and how to find the issue in one sentence.

"It is possible to have a typo in `@Named` value when providing and injecting the dependency, we can find the issue after building the project when Dagger graph is generated."